### PR TITLE
TandemFoil: lr=2e-4 + T_max=5 (ultra-rapid cosine cycling)

### PR DESCRIPTION
**Pull request by tcapelle**
## Hypothesis
Ultra-rapid cosine annealing with T_max=5 (vs current best T_max=10) may help the Lion optimizer escape sharp minima more aggressively. Rapid cycling can act as implicit regularization, preventing overfitting to local surface-pressure patterns. The current best achieved 45.07 MAE with T_max=10 — cutting the period in half doubles the number of restarts, potentially exploring more of the loss landscape.

## Instructions
Run the following command exactly:

```bash
cd target/icml2026 && python train.py \
  --dataset tandemfoil \
  --optimizer lion \
  --lr 2e-4 \
  --cosine-t-max 5 \
  --no-use-ema \
  --model-slices 64 \
  --enable-fourier \
  --enable-te-coord-frame \
  --enable-cp-panel \
  --enable-cp-panel-tandem-only \
  --asinh-pressure \
  --residual-prediction \
  --enable-pressure-prior-addition \
  --epochs 999 \
  --wandb-group tandem-lr2e4-compounds \
  --wandb-name "historia/tandem-lr2e4-tmax5"
```

Report the best `val_primary/surface_pressure_mae` achieved and the W&B run ID.

## Baseline
Current best metrics (PR #2610):
- val_primary/surface_pressure_mae: **45.07**
- W&B run: `ixs1rqgk`
- Epochs: 119
- Reproduce: `cd target/icml2026 && python train.py --dataset tandemfoil --optimizer lion --lr 2e-4 --cosine-t-max 10 --no-use-ema --model-slices 64 --enable-fourier --enable-te-coord-frame --enable-cp-panel --enable-cp-panel-tandem-only --asinh-pressure --residual-prediction --enable-pressure-prior-addition --epochs 999`

### CODE REVIEW

**Comment by tcapelle** [comment]
STUDENT historia:

## Results

### Key Metrics

| Metric | Baseline (PR #2610, 119 ep) | This Run (Best Val) | This Run (Final, Ep 117) |
|--------|---------------------------|---------------------|--------------------------|
| val_primary/surface_pressure_mae | **45.07** | 45.22 (ep 101) | 52.64 |
| test_primary/surface_pressure_mae | — | — | 59.65 |

### Training Summary

- **Total epochs completed:** 117 (timeout)
- **W&B run ID:** `mk6r5tw5` ([link](https://wandb.ai/wandb-applied-ai-team/senpai-v1/runs/mk6r5tw5))
- **Command used:**
  ```bash
  cd target/icml2026 && CUDA_VISIBLE_DEVICES=0 python train.py \
    --dataset tandemfoil --optimizer lion --lr 2e-4 --cosine-t-max 5 \
    --no-use-ema --model-slices 64 --enable-fourier --enable-te-coord-frame \
    --enable-cp-panel --enable-cp-panel-tandem-only --asinh-pressure \
    --residual-prediction --enable-pressure-prior-addition \
    --epochs 999 --wandb-group tandem-lr2e4-compounds \
    --wandb-name "historia/tandem-lr2e4-tmax5" --agent historia
  ```

### Per-Epoch Trajectory (key milestones)

| Epoch | Val MAE | Loss | Notes |
|-------|---------|------|-------|
| 1 | 189.25 | 0.899 | Start |
| 10 | 90.39 | 0.193 | |
| 20 | 69.37 | 0.156 | |
| 28 | 58.97 | 0.155 | |
| 38 | 55.43 | 0.143 | |
| 49 | 54.69 | 0.124 | |
| 52 | 50.30 | 0.128 | First sub-55 |
| 66 | 48.55 | 0.111 | |
| 86 | 47.96 | 0.089 | |
| 96 | 45.37 | 0.086 | Near baseline |
| 101 | **45.22** | 0.096 | **BEST VAL** — 0.15 above baseline |
| 117 | 52.64 | 0.110 | Final (timeout) |

### What Happened

**The hypothesis did not beat the baseline.** T_max=5 (ultra-rapid cosine cycling) produced a best val of 45.22, which is essentially tied with the 45.07 baseline but did not improve upon it.

**Key observations:**

1. **Extreme validation variance** — the T_max=5 cycling created wild oscillations in val MAE, ranging from 45-85 across epochs. The baseline with T_max=10 likely had much smoother convergence.

2. **Slower convergence** — at equivalent epoch count (119 vs 117), the T_max=5 run barely matched the baseline. The rapid cycling appears to prevent the model from settling into good minima before being perturbed again.

3. **Test metric is poor** — test_primary at 59.65 (evaluated at final ep 117) is much worse than the best val, highlighting the instability. The model happened to be at an unfavorable point in the cosine cycle at timeout.

4. **Note: Infinity in some test metrics** — `test_geom_camber_cruise/volume_mae` reported Infinity, which may indicate a numerical issue with volume predictions on that geometry split.

5. **T_max=5 is too aggressive** — the hypothesis that more frequent restarts would act as implicit regularization and explore more of the loss landscape did not pan out. The restarts are too frequent for the model to converge between cycles.

### Suggested Follow-ups

1. **T_max=7 or T_max=8** — a middle ground between T_max=5 (too aggressive) and T_max=10 (baseline) might find a sweet spot.
2. **Decaying LR restarts** — reduce the peak LR in later cycles to allow the model to settle more as training progresses, while still benefiting from early exploration.
3. **Longer T_max with more epochs** — T_max=20 or T_max=30 with a longer budget could give smoother convergence and potentially better results.

**Comment by tcapelle** [comment]
ADVISOR: Closing PR #2729 because ADVISOR: Closing — 45.219 at T_max=5 is essentially tied with baseline 45.07 (+0.3%). T_max=5 doesn't improve TandemFoil at lr=2e-4. T_max=10 remains optimal..
